@xen-orchestra/web-core/lib/components/UiChip.vue

```diff
@@ -0,0 +1,181 @@
+    --remove-chip-color: var(--color-green-base);
+
+    &:is(:hover, .hover, :focus-visible) {
+      --color: var(--color-grey-100);
+      --icon-color: var(--color-grey-200);
+      --remove-chip-color: var(--color-green-d20);
+      --background-color: var(--background-color-green-20);
+    }
+
+    &:is(:active, .pressed) {
+      --color: var(--color-grey-100);
+      --icon-color: var(--color-grey-200);
+      --remove-chip-color: var(--color-green-d40);
+      --background-color: var(--background-color-green-30);
+    }
+
+    &.disabled {
+      --color: var(--color-grey-600);
+      --icon-color: var(--color-grey-600);
+      --remove-chip-color: var(--color-grey-600);
+      --background-color: var(--color-green-l60);
+    }
+  }
+
+  &.warning {
+    --color: var(--color-grey-100);
+    --icon-color: var(--color-grey-200);
+    --background-color: var(--background-color-orange-10);
+    --remove-chip-color: var(--color-orange-base);
+
+    &:is(:hover, .hover, :focus-visible) {
+      --color: var(--color-grey-100);
+      --icon-color: var(--color-grey-200);
+      --remove-chip-color: var(--color-orange-d20);
+      --background-color: var(--background-color-orange-20);
+    }
+
+    &:is(:active, .pressed) {
+      --color: var(--color-grey-100);
+      --icon-color: var(--color-grey-200);
+      --remove-chip-color: var(--color-orange-d40);
+      --background-color: var(--background-color-orange-30);
+    }
+
+    &.disabled {
+      --color: var(--color-grey-600);
+      --icon-color: var(--color-grey-600);
+      --remove-chip-color: var(--color-grey-600);
+      --background-color: var(--color-orange-l60);
+    }
+  }
+
+  &:is(.danger, .error) {
+    --color: var(--color-grey-100);
+    --icon-color: var(--color-grey-200);
+    --background-color: var(--background-color-red-10);
+    --remove-chip-color: var(--color-red-base);
+
+    &:is(:hover, .hover, :focus-visible) {
+      --color: var(--color-grey-100);
+      --icon-color: var(--color-grey-200);
+      --remove-chip-color: var(--color-red-d20);
+      --background-color: var(--background-color-red-20);
+    }
+
+    &:is(:active, .pressed) {
+      --color: var(--color-grey-100);
+      --icon-color: var(--color-grey-200);
+      --remove-chip-color: var(--color-red-d40);
+      --background-color: var(--background-color-red-30);
+    }
+
+    &.disabled {
+      --color: var(--color-grey-600);
+      --icon-color: var(--color-grey-600);
+      --remove-chip-color: var(--color-grey-600);
+      --background-color: var(--color-red-l60);
+    }
+  }
+}
+
+/* IMPLEMENTATION */
+.ui-chip {
+  display: inline-flex;
+  align-items: center;
+  gap: 0.8rem;
+  padding: 0.4rem 0.8rem;
+  border-radius: 10rem;
+  color: var(--color);
+  background-color: var(--background-color);
+  cursor: pointer;
+
+  &.disabled {
+    pointer-events: none;
+  }
+
+  .content {
+    display: inline-flex;
+    align-items: center;
+    gap: 0.8rem;
+    white-space: nowrap;
+  }
```

Comment:
- Removing no longer needed styles
- Adding a line-height trick for better vertical alignment of the label
- Adding overflow and text-overflow for better handling of limited size

```suggestion
  .content {
    line-height: 0.9;
    overflow: hidden;
    text-overflow: ellipsis;
  }
```